sigma/backends/loki/loki.py

```diff
@@ -561,19 +561,20 @@ def find_longest_common_string_line_filter(
             return None
         matcher = None
         match = None
-        value = None
+        value: Optional[str] = None
         # Finding the longest common substring of a list of strings, by repeatedly
         # calling SequenceMatcher's find_longest_match. The 1st candidate is cached
         # in the 2nd sequence (b), then following candidates are set as the 1st
         # sequence (a). The longest match between a and b is then found, each time
         # reducing the search region of b based on the previous match.
         # See: https://docs.python.org/3/library/difflib.html#difflib.SequenceMatcher
         for cand in candidates:
-            if matcher is None:
+            if cand is None:

```

Comment:
If we are checking for `None` only, your code works, but if we want to check for empty list, False, and any other zero value, this is the way to do that.
```suggestion
            if not cand:
```

Reply:
If we don't trust mypy's type verification, why use it? `candidates` is _supposed_ to be a `List[Optional[LogQLLineFilterInfo]]` - so yes, it _should_ only be `None` or a valid object.

Reply:
Mypy is not perfect and has misses.